Segfault when using the PSI example with YAO sharing

The example ./psi_phasing_test -r 1 -n 1 -b 32 -g 1 (YAO sharing) produces a segmentation fault when the framework is compiled in release mode. The segmentation fault does not occur when the framework is built in debug mode.

compiler: gcc (GCC) 8.2.1 20181127
debugger: GNU gdb (GDB) 8.2.1
cmake version 3.13.4

Hi,

I think the reason of the crashes are the missing return statements of `gen_entries` and `gen_cuckoo_entries`. GCC and Clang already warned:
```
/home/lennart/uni/Encrypto/ABY/ABY/src/examples/psi_phasing/common/hashing/simple_hashing.cpp: In function ‘void* gen_entries(void*)’:
/home/lennart/uni/Encrypto/ABY/ABY/src/examples/psi_phasing/common/hashing/simple_hashing.cpp:126:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
 ^
```
```
/path/to/ABY/src/examples/psi_phasing/common/hashing/simple_hashing.cpp:126:1: warning: 
      control reaches end of non-void function [-Wreturn-type]
}
```

Also, when compiled (even in Debug mode) with Clang, the program crashes with a SIGILL, which is often a good indicator for such problems:
```
Core was generated by `./bin/psi_phasing_test -r 0 -n 1 -b 32 -g 1'.
Program terminated with signal SIGILL, Illegal instruction.
#0  0x000055d4781547d7 in gen_entries (ctx_tmp=0x55d4795b6860) at /path/to/ABY/src/examples/psi_phasing/common/hashing/simple_hashing.cpp:124
124		free(address);
```

A `return nullptr;` at the end of the functions should be sufficient to fix this.

--- a/src/examples/psi_phasing/common/hashing/cuckoo.cpp
+++ b/src/examples/psi_phasing/common/hashing/cuckoo.cpp
@@ -174,6 +174,7 @@ void *gen_cuckoo_entries(void *ctx_void) {
 	for(i = ctx->startpos; i < ctx->endpos; i++, eleptr+=inbytelen) {
 		gen_cuckoo_entry(eleptr, ctx->cuckoo_entries + i, hs, i);
 	}
+	return nullptr;
 }
 
 

--- a/src/examples/psi_phasing/common/hashing/simple_hashing.cpp
+++ b/src/examples/psi_phasing/common/hashing/simple_hashing.cpp
@@ -121,6 +121,7 @@ void *gen_entries(void *ctx_tmp) {
 	}
 	free(tmpbuf);
 	free(address);
+	return nullptr;
 }
 
 inline void insert_element(sht_ctx* table, uint8_t* element, uint32_t* address, uint8_t* tmpbuf, hs_t* hs) {

